src/exportJarFileCommand.ts

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { EOL, platform } from "os";
+import { basename, extname, join } from "path";
+import { CancellationToken, commands, Extension, extensions, ProgressLocation,
+         QuickInputButtons, QuickPick, QuickPickItem, Uri, window, workspace } from "vscode";
+import { buildWorkspace } from "./build";
+import { isStandardServerReady } from "./extension";
+import { Jdtls } from "./java/jdtls";
+import { INodeData } from "./java/nodeData";
+import { WorkspaceNode } from "./views/workspaceNode";
+
+enum ExportSteps {
+    ResolveProject = "RESOLVEPROJECT",
+    ResolveMainMethod = "RESOLVEMAINMETHOD",
+    GenerateJar = "GENERATEJAR",
+    Finish = "FINISH",
+}
+
+export async function createJarFile(node?: INodeData) {
```

Comment:
I observed some UX issues here:

1. The status report in the status bar looks strange to me. Because at this point, there is no pending requests actually, but it showing the spinning icon.
![image](https://user-images.githubusercontent.com/6193897/88260397-e0936a80-ccf6-11ea-9c05-ae70d6e22254.png)


2. When you are in the middle of the command process, if you click export jar again, it will compile the workspace, which seems not correct

3. When I try to export jar for spring pet clinic, it takes very long time to resolve the main method.(actually it never ends) why?

4. No need to report error when user canceled
![image](https://user-images.githubusercontent.com/6193897/88260740-952d8c00-ccf7-11ea-9e5a-c2be3e783a5a.png)